### PR TITLE
DEVOPS-2114: Configure CI for Java SDK

### PR DESCRIPTION
**Pull request by daniilyar-incountry**
This PR implements the following:

- Execute Sonar scanner on each CI run. For every branch and for every PR
- Execute Snyk scan should be executed for commits on master; SB_* ; RC_* ; develop  branches
- Block the PR if the Sonar quality gate fails

Example links:
- Example Travis build: https://travis-ci.com/github/incountry/sdk-java/builds/182474574
- Example Sonar report for PR: https://sonarcloud.io/dashboard?id=incountry_sdk-java&pullRequest=154 
- Example Sonar report for branch: https://sonarcloud.io/dashboard?id=incountry_sdk-java&branch=FB_DEVOPS-2114
- Example SNYK report: https://app.snyk.io/org/incountry/project/8021f514-b208-422e-9157-af6a445e9f50/history/f6c268c9-ca73-49a2-924c-1e6ed1595898
- Example coverage report: https://sonarcloud.io/component_measures?branch=FB_DEVOPS-2114&id=incountry_sdk-java&metric=new_lines_to_cover&view=list

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,27 +13,46 @@ sudo: false
...
+        # Get gradle zip file name and gradle directory name
+        - arr=(${gradleUrl//// }); gradleZipFile=${arr[${#arr[*]}-1]}
+        - arr2=(${gradleZipFile//-bin/ }); gradleDirectory=$arr2
+        # Check if Gradle binaries exist. If not, download them
+        - if [ -d "$PWD/gradle-container/$gradleDirectory" ]; then mv $PWD/gradle-container/$gradleDirectory $PWD; else wget -q ${gradleUrl}; unzip -qq ${gradleZipFile}; fi
+        - export GRADLE_HOME=$PWD/${gradleDirectory}; export PATH=$GRADLE_HOME/bin:$PATH
+        - gradle -v
+        # We need target branch and master branch to be present in order to allow Sonar properly build test coverage difference with current branch, master branch and target branch
+        - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then git fetch --no-tags https://github.com/${TRAVIS_PULL_REQUEST_SLUG}.git +refs/heads/${TRAVIS_BRANCH}:refs/remotes/origin/${TRAVIS_BRANCH}; fi
+        - if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then git fetch --no-tags https://github.com/${TRAVIS_PULL_REQUEST_SLUG}.git +refs/heads/master:refs/remotes/origin/master; fi
```
Comment (VladimirLevitskiy) [inline .travis.yml:43]: This row is not required

File: travis-build.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+# Return true if branch matches the grep regexp pattern specified and false otherwise
+branch_matches() {
+  if grep -qE "$1" <(echo "$TRAVIS_BRANCH"); then return 0; else return 1; fi
+}
+
+# Perform the build
+gradle build jacocoTestReport sonarqube
+gradle integrationTest
```
Comment (VladimirLevitskiy) [inline travis-build.sh:28]: Is it possible to make parallel builds for unit tests+ sonar scan and integration tests? For speed up builds

**Review by VladimirLevitskiy — CHANGES_REQUESTED** [review]

File: travis-build.sh (hunk 2 above)
Comment (daniilyar-incountry) [reply travis-build.sh:28]: @VladimirLevitskiy sure, great idea! :) if you are sure that tests support parallel builds and will not deadlock on smth or fight for test resources :)

**Review by VladimirLevitskiy — APPROVED** [review]
LGTM

**Review by RommeroInCountry — APPROVED** [review]
LGTM

File: travis-build.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then # Fetch the PR branch with complete history for Travis PR builds in order to let Sonar properly display annotations in coverage details
+  git fetch --no-tags https://github.com/${TRAVIS_PULL_REQUEST_SLUG}.git +refs/heads/${TRAVIS_BRANCH}:refs/remotes/origin/${TRAVIS_BRANCH}
+fi
+
+# Perform the build
+gradle build jacocoTestReport sonarqube
+
+if [[ "$TRAVIS_PULL_REQUEST" == 'false' ]] && branch_matches "^master$|^develop$|^SB_*|^RC_*"; then
+  npm install -g snyk
+  snyk monitor --org=incountry --prune-repeated-subdependencies --remote-repo-url="${APP_NAME}" --project-name="${APP_NAME}:${TRAVIS_BRANCH}"
```
Comment (RommeroInCountry) [inline travis-build.sh:36]: Does java snyk also use node.js implementation?

**Review by samutaa — APPROVED** [review]

**Pull request merged**